### DISCUSSION TITLE
Ten threads using the same connection is very slow

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#18106] [Q&A and General discussion] [answered]
Milvus：v2.0.2
Pymilvus：2.0.2

A total of 100 threads are created, and a new connection is created for every 10 threads. Even after 10 connections are created, I can still see the speed getting slower and slower. Speeds range from 27ms to 150ms.

But each thread creates a new connection at a normal speed of around 15ms.

Create one connection for every 10 processes —— Speeds range from 27ms to 150ms. (Most range from 70ms to 100ms.
Create one connection for every 5   processes —— Speeds range from 20ms to 120ms. (Most range from 60ms to 100ms.
Create one connection for every 3   processes —— Speeds range from 16ms to  60ms.  (Most range from 35ms to 50ms.
Create one connection for every 1   processes —— Speeds range from 10ms to  20ms.  (Most range from 12ms to 16ms.

So why does every 10 threads using a new connection still slow down? Are the queries in the same connection serial?

**Comment by yhmo** [comment, marked as answer]
In v2.0.2, search requests are put into a queue, and executed one by one.
10 threads share one connection, the performance is worst. I guess is because of multi-threads competition, not sure.

**Reply by xiaofan-luan** [reply]
This is not a issue in 2.1, we will try to combine requests and do search in parallel
